Add additional filters: `datetime_from_period` and `merge_date_url`

minchinweb opened the pull request:
A couple of tags I need to expand my [Seafoam theme](https://github.com/MinchinWeb/seafoam) to support period archive pages.

Also adds tests for all included filters. Also adds running tests to the GitHub workflow.

minchinweb committed:
[datetime_from_period] first pass

minchinweb committed:
[datetime_from_period] Allow passing of a tuple of three integers to get a datetime.

minchinweb committed:
[merge_date_url] first pass

minchinweb committed:
[test] very first tests

minchinweb committed:
[GitHub] add test running to GitHub workflow

minchinweb committed:
[article_date] switch to more friendly cross-platform formatting

minchinweb committed:
[test] add full suite

MinchinWeb commented:
I'm not sure why the GitHub tests aren't running...

The code here is probably good to go, but I haven't actually done the theme development with them yet.... So no current rush to merge this while the other Pull Requests are still outstanding...

minchinweb committed:
[Readme] update with new filters

justinmayer commented:
@MinchinWeb: Now that #8 has been merged, you'll need to rebase this PR onto `master`, resolving any conflicts in `tasks.py`.

minchinweb committed:
[datetime_from_period] allow passing a straight integer (assumed to be year)

minchinweb committed:
[GitHub] re-enable GitHub Workflows

add `--diff` option to `invoke lint`, upgrade black

MinchinWeb commented:
@justinmayer We should be good to merge to master and cut a release.

I've rebased to master (and discovered/made use of some of what you'd mentioned earlier about [interactive rebasing](https://github.com/pelican-plugins/jinja-filters/pull/8#issuecomment-826326231)), fixed a few remaining bugs, and tested it locally, and everything looks good.

I got the GitHub Workflows working again although I had a fair bit of trouble (since resolved) with the linters as the version of black installed by poetry didn't match the version used by pre-commit. To help with that, I added a `--diif` pass-through option for `invoke lint` and enabled it for the Workflow.

justinmayer commented:
@MinchinWeb: Do you want to update this PR with Black 21.4b0, which was just released 12 hours ago? 😅

minchinweb committed:
update Changelog

minchinweb committed:
[requirements] update to lastest black version
black version 21.4b0, update minimum Python version to 3.6.2

MinchinWeb commented:
@justinmayer *black* is updated! (It also requires bumping the minimum version of Python from 3.6.0 to 3.6.2).

MinchinWeb commented:
(I think black is particularly prone to this because almost every update is an update to the "major" version number, which keeps poetry from automatically updating it.)

justinmayer commented on CHANGELOG.rst:
```diff
@@ -1,6 +1,12 @@
+- **support** rework supporting ``tasks.py`` (for use with ``invoke``)
+- **support** add test suite
+- **support** support Pelican from version 3.0 on
+
```
The core design philosophy behind AutoPub includes never needing to manually add entries to the change-log. These entries should instead be in the form of a `RELEASE.md` file:
```markdown
Release type: patch

- **feature** add ``merge_date_url`` filter
- **feature** add ``datetime_from_period`` filter
- **support** rework supporting ``tasks.py`` (for use with ``invoke``)
- **support** add test suite
- **support** support Pelican from version 3.0 on
```

minchinweb replied:
@justinmayer Done! I've moved this to a `RELEASE.md` file.

justinmayer reviewed (approved):
Thanks for all the enhancements! ✨
